### PR TITLE
Add new test 'Change position for attributes and values'

### PR DESCRIPTION
**Pull request by boubkerbribri**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.7.x
| Description?  | Add new test 'Change position for attributes and values'
| Type?         | refacto
| Category?     | TE
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | no
| How to test?  | `URL_FO=shopUrl/ TEST_PATH='functional/BO/03_catalog/04_attributesAndFeatures/01_attributes/06_changeAttributePosition' npm run specific-test`

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/21324)

### CODE REVIEW

File: tests/UI/campaigns/functional/BO/03_catalog/04_attributesAndFeatures/01_attributes/06_changeAttributePosition.js [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+
+const baseContext = 'functional_BO_shopParameters_catalog_attributesAndFeatures_attributes_changeAttributePosition';
+
+// Import expect from chai
+const {expect} = require('chai');
+
+// Browser and tab
+let browserContext;
+let page;
+
```
Comment (SimonGrn) [inline tests/UI/campaigns/functional/BO/03_catalog/04_attributesAndFeatures/01_attributes/06_changeAttributePosition.js:28]: No comment to explain the test ?

File: tests/UI/campaigns/functional/BO/03_catalog/04_attributesAndFeatures/01_attributes/06_changeAttributePosition.js [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+const attributesPage = require('@pages/BO/catalog/attributes');
+const viewAttributePage = require('@pages/BO/catalog/attributes/view');
+
+// Import data
+const {Attributes} = require('@data/demo/attributes');
+
+// Import test context
+const testContext = require('@utils/testContext');
+
+const baseContext = 'functional_BO_shopParameters_catalog_attributesAndFeatures_attributes_changeAttributePosition';
```
Comment (Progi1984) [inline tests/UI/campaigns/functional/BO/03_catalog/04_attributesAndFeatures/01_attributes/06_changeAttributePosition.js]:
```suggestion
const baseContext = 'functional_BO_catalog_attributesAndFeatures_attributes_changeAttributePosition';
```

**Review by Progi1984 — COMMENTED** [review]
A small error

File: tests/UI/campaigns/functional/BO/03_catalog/04_attributesAndFeatures/01_attributes/06_changeAttributePosition.js [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+      nonSortedTable = await nonSortedTable.map(text => parseFloat(text));
+      sortedTable = await sortedTable.map(text => parseFloat(text));
+
+      const expectedResult = await attributesPage.sortArray(nonSortedTable, true);
+
+      await expect(sortedTable).to.deep.equal(expectedResult);
+    });
+
+    it('should change first attribute position to 3', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'changePosition', baseContext);
```
Comment (nesrineabdmouleh) [inline tests/UI/campaigns/functional/BO/03_catalog/04_attributesAndFeatures/01_attributes/06_changeAttributePosition.js]:
```suggestion
      await testContext.addContextItem(this, 'testIdentifier', 'changeAttributePosition', baseContext);
```

File: tests/UI/campaigns/functional/BO/03_catalog/04_attributesAndFeatures/01_attributes/06_changeAttributePosition.js [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+      const textResult = await attributesPage.changePosition(page, 1, 3);
+      await expect(textResult, 'Unable to change position').to.contains(attributesPage.successfulUpdateMessage);
+
+      // Get third row attribute name and check if is equal the first row attribute name before changing position
+      const thirdRowAttributeName = await attributesPage.getTextColumn(page, 3, 'b!name');
+      await expect(thirdRowAttributeName, 'Changing position was done wrongly').to.equal(firstRowAttributeName);
+    });
+
+    it('should reset third attribute position to 1', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'resetPosition', baseContext);
```
Comment (nesrineabdmouleh) [inline tests/UI/campaigns/functional/BO/03_catalog/04_attributesAndFeatures/01_attributes/06_changeAttributePosition.js]:
```suggestion
      await testContext.addContextItem(this, 'testIdentifier', 'resetAttributePosition', baseContext);
```

File: tests/UI/campaigns/functional/BO/03_catalog/04_attributesAndFeatures/01_attributes/06_changeAttributePosition.js [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+    // Should reset filters before changing position
+    it('should reset all filters and get number of attributes in BO', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'resetFilterFirst', baseContext);
+
+      const numberOfValues = await viewAttributePage.resetAndGetNumberOfLines(page);
+      await expect(numberOfValues).to.be.above(2);
+    });
+
+    it('should change first value position to 3', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'changePosition', baseContext);
```
Comment (nesrineabdmouleh) [inline tests/UI/campaigns/functional/BO/03_catalog/04_attributesAndFeatures/01_attributes/06_changeAttributePosition.js]:
```suggestion
      await testContext.addContextItem(this, 'testIdentifier', 'changeValuePosition', baseContext);
```

File: tests/UI/campaigns/functional/BO/03_catalog/04_attributesAndFeatures/01_attributes/06_changeAttributePosition.js [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+      const textResult = await viewAttributePage.changePosition(page, 1, 3);
+      await expect(textResult, 'Unable to change position').to.contains(attributesPage.successfulUpdateMessage);
+
+      // Get third row attribute name and check if is equal the first row attribute name before changing position
+      const thirdRowValueName = await viewAttributePage.getTextColumn(page, 3, 'b!name');
+      await expect(thirdRowValueName, 'Changing position was done wrongly').to.equal(firstRowValueName);
+    });
+
+    it('should reset third value position to 1', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'resetPosition', baseContext);
```
Comment (nesrineabdmouleh) [inline tests/UI/campaigns/functional/BO/03_catalog/04_attributesAndFeatures/01_attributes/06_changeAttributePosition.js]:
```suggestion
      await testContext.addContextItem(this, 'testIdentifier', 'resetValuePosition', baseContext);
```

File: tests/UI/campaigns/functional/BO/03_catalog/04_attributesAndFeatures/01_attributes/06_changeAttributePosition.js (hunk 2 above)
Comment (boubkerbribri) [reply tests/UI/campaigns/functional/BO/03_catalog/04_attributesAndFeatures/01_attributes/06_changeAttributePosition.js]: Thanks for noticing this, there was the same error on all tests in this directory.

File: tests/UI/campaigns/functional/BO/03_catalog/04_attributesAndFeatures/01_attributes/06_changeAttributePosition.js (hunk 3 above)
Comment (boubkerbribri) [reply tests/UI/campaigns/functional/BO/03_catalog/04_attributesAndFeatures/01_attributes/06_changeAttributePosition.js]: done

File: tests/UI/campaigns/functional/BO/03_catalog/04_attributesAndFeatures/01_attributes/06_changeAttributePosition.js (hunk 4 above)
Comment (boubkerbribri) [reply tests/UI/campaigns/functional/BO/03_catalog/04_attributesAndFeatures/01_attributes/06_changeAttributePosition.js]: done

File: tests/UI/campaigns/functional/BO/03_catalog/04_attributesAndFeatures/01_attributes/06_changeAttributePosition.js (hunk 5 above)
Comment (boubkerbribri) [reply tests/UI/campaigns/functional/BO/03_catalog/04_attributesAndFeatures/01_attributes/06_changeAttributePosition.js]: done

File: tests/UI/campaigns/functional/BO/03_catalog/04_attributesAndFeatures/01_attributes/06_changeAttributePosition.js (hunk 6 above)
Comment (boubkerbribri) [reply tests/UI/campaigns/functional/BO/03_catalog/04_attributesAndFeatures/01_attributes/06_changeAttributePosition.js]: done

File: tests/UI/campaigns/functional/BO/03_catalog/04_attributesAndFeatures/01_attributes/06_changeAttributePosition.js (hunk 1 above)
Comment (boubkerbribri) [reply tests/UI/campaigns/functional/BO/03_catalog/04_attributesAndFeatures/01_attributes/06_changeAttributePosition.js:28]: The describe say all what needed to know, no need for further explaining

File: tests/UI/campaigns/functional/BO/03_catalog/04_attributesAndFeatures/01_attributes/06_changeAttributePosition.js (hunk 2 above)
Comment (boubkerbribri) [reply tests/UI/campaigns/functional/BO/03_catalog/04_attributesAndFeatures/01_attributes/06_changeAttributePosition.js]: done, thanks

File: tests/UI/campaigns/functional/BO/03_catalog/04_attributesAndFeatures/01_attributes/06_changeAttributePosition.js (hunk 1 above)
Comment (boubkerbribri) [reply tests/UI/campaigns/functional/BO/03_catalog/04_attributesAndFeatures/01_attributes/06_changeAttributePosition.js:28]: Description added to the test

**Review by Progi1984 — APPROVED** [review]

**Review by SimonGrn — APPROVED** [review]

**Pull request merged**